### PR TITLE
Issue 409 second attempt

### PR DESCRIPTION
**Pull request by mortenjc**
### Issue reference / description

The branch you merge from should already reference an event-formation-unit github ticket number. You can add a descriptive title, but if an issue is referenced, you don't have to.

## Checklist for submitter

- [ ] Check for conflict with integration test
- [ ] Unit tests pass

---

## Nominate for Group Code Review (Anyone can nominate it)
Indicate if you think the code should be reviewed in a Thursday code review session.

- [ ] Recommend for group code review

Also, nominate it on the code_review Slack channel.

### CODE REVIEW

**Comment by mortenjc** [comment]
I do not understand why the system tests fail. I can't see any changes that would cause this, but then
again the error messages are not crystal clear.

**Comment by mortenjc** [comment]
In any case I have added 

- freia module which uses the ESS parser and VMM3 parser
- VMM3Parser with unit tests
- Moved ReadoutParser and ESSTime in to readout/common and VMM3Parser into readout/vmm3
- The move affected DREAM and LoKI (mainly include and CMakelists)
- The freia module does not yet produce any events. This will come in later PRs

**Review by amues — APPROVED** [review]
The pull request looks good. I left some comments, feel free to address them if relevant and merge.

File: src/modules/freia/FreiaBase.cpp [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+  };
+  EV42Serializer Serializer{KafkaBufferSize, "freia", Produce};
+
+  unsigned int DataIndex;
+  TSCTimer ProduceTimer(EFUSettings.UpdateIntervalSec * 1000000 * TSC_MHZ);
+  Timer h5flushtimer;
+  // Monitor these counters
+  RuntimeStat RtStat({Counters.RxPackets, Counters.Events, Counters.TxBytes});
+
+  while (runThreads) {
```
[the remark this answers is not in the document]
Comment (mortenjc) [reply src/modules/freia/FreiaBase.cpp:184]: Good catch. I was only making changes to the processing thread so I overlooked this.

File: src/modules/readout/vmm3/VMM3Parser.cpp [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+  if (Buffer == nullptr) {
+    Stats.ErrorSize++;
+    XTRACE(DATA, WAR, "Invalid data pointer");
+    return GoodReadouts;
+  }
+
+  if (Size % DataLength != 0) {
+    Stats.ErrorSize++;
+    XTRACE(DATA, WAR, "Invalid data length - %d should be multiple of 20", Size);
```
[the remark this answers is not in the document]
Comment (mortenjc) [reply src/modules/readout/vmm3/VMM3Parser.cpp:31]: Yes!

File: src/modules/readout/vmm3/VMM3Parser.cpp [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    }
+
+    if ((Readout.FENId > MaxFENId) or (Readout.FENId == 0))  {
+      XTRACE(DATA, WAR, "Invalid FENId %d (valid: 1 - %d)", Readout.FENId, MaxFENId);
+      Stats.ErrorFEN++;
+      continue;
+    }
+
+    if (Readout.DataLength != DataLength)  {
+      XTRACE(DATA, WAR, "Invalid header length - must be 20 bytes", Readout.DataLength);
```
[the remark this answers is not in the document]
Comment (mortenjc) [reply src/modules/readout/vmm3/VMM3Parser.cpp:52]: Yes, but here you missed a bug ;-)  I forgot to add a %d to XTRACE

**Pull request merged**